submissions/Barasii/a-tiny-JS-world-OOP/index.js

```diff
@@ -0,0 +1,86 @@
+class Inhabitant {
+   constructor(species, name, gender, legs, hands, saying, friends) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.hands = hands;
+      this.saying = saying;
+      this.friends = [];
+   }
+
+   toString() {
+      return Object.keys(this).map(prop => `${this[prop]}`).join('; ');
+      }
+
+   callFriends(friends) {         
+      this.friends = friends;
+      }
+}
+
+const Meow = superclass =>
+   class extends superclass {
+      constructor(species, name, gender, legs, hands, saying, friends) {
+         super(species, name, gender, legs, hands, saying, friends);
+         this.saying = 'mur-mur';
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, friends) {
+      super('human', name, gender, 2, 2, saying, friends);
+   }
+}
+
+class Man extends Human {
+   constructor(name, saying, friends) {
+      super(name, 'male', saying, friends);
+   }
+}
+
+class Woman extends Human {
+   constructor(name, saying, friends) {
+      super(name, 'female', saying, friends);
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, saying, friends) {
+      super(species, name, gender, 4, 0, saying, friends);
+   }
+}
+
+class Dog extends Animal {
+   constructor( name, gender, saying, friends) {
+      super('dog', name, gender, saying, friends);
+   }
+}
+
+class Cat extends Meow(Animal) {
+   constructor(name, gender, friends) {
+      super('cat', name, gender, friends);
+   }
+}
+
+class CatWoman extends Meow(Human) {
+   constructor(name, friends) {
+      super(name, 'female', friends);
+   }
+}
+
+const man = new Man('Petro', 'Zdorov!'),
+      woman = new Woman('Katia', 'Pryvit)'),
+      dog = new Dog('Nika','female', 'gav-gav'),
+      cat = new Cat('Musia', 'female'),
+      catWoman = new CatWoman('Murka');
```

Comment:
- [ ] All inhabitants are stored in a container (array or object)

It's a challenge to refer them later when you're assigning friends. But doable.
Why do we need a place where they all collected? It's about a data structure. You will benefit from grouping objects properly. It pays with a better control. Imagine, you need to pass all inhabitants to a different module when scaling your application.

Reply:
I hope I have done well